Gh file download

jim-sheldon opened the pull request:
Not done yet, but want eyes sooner for a faster merge.
The `case.py` file should become its own package so we can `import` it.

jim-sheldon commented:
Taking this out of draft with the caveats that I will add error handling, ensure tests pass, and squash/fixup commits before merging.

abhidg commented on scripts/gh_data_update/setup.py:
```diff
@@ -0,0 +1,111 @@
+	database = client[DATABASE_NAME]
+	logging.info("Creating collections")
+	_ = database[CDC_COLLECTION]
+	_ = database[WHO_COLLECTION]
```
create the Gh collection here too?

jim-sheldon replied:
Yeah. Pymongo access and creation are the same, which is weird, so technically these LOCs don't do anything, but it is good to be explicit.

abhidg commented on scripts/gh_data_update/test_run.py:
```diff
@@ -0,0 +1,58 @@
+
+
+def csv_to_dict(data: str) -> list[dict]:
+	rows = "\n".join([f"{','.join(d)}" for d in data])
+	f = io.StringIO(rows)
```
data is list[str], are we sure there are no commas in the data?

abhidg reviewed (commented):
Few changes, otherwise looks good -- are we going to deprecate src/app.py? Looks like this will do most of the work, minus the aggregation into total counts.

jim-sheldon commented on scripts/gh_data_update/run.py:
```diff
@@ -0,0 +1,214 @@
+		collection.insert_one(case_dict)
+
+
+def remove_cases(count, location):
```
That opens up a whole separate set of issues I would rather avoid. Given our intended use for this line list should not include values other than location, case status, and whatever else we use, I think we can forego that check for now. In the future I agree we will need that.

jim-sheldon commented:
Unless there are any blockers, I would like to punt on addressing comments until a future PR, so I can get it up and running ASAP. Refinements should happen but barring anything glaring I think can wait until after crunch time.

jim-sheldon commented on scripts/gh_data_update/run.py:
```diff
@@ -0,0 +1,214 @@
+	try:
+		collection = get_db_collection(GH_COLLECTION)
+		cursor = collection.find({})
+		return cases_to_csv([case for case in cursor])
```
I think passing data as inputs and outputs makes for a cleaner design, and avoids passing open connections between functions.

abhidg replied:
Fine for the monkeypox use case, as we are less than 100k cases, but this will break down at large numbers as the entire list needs to be loaded into memory.

jim-sheldon commented:
> Few changes, otherwise looks good -- are we going to deprecate src/app.py? Looks like this will do most of the work, minus the aggregation into total counts.

Yeah, we should be able to move everything into the scripts folder.
